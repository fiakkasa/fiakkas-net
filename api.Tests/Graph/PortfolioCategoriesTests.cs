using HotChocolate.Execution;

namespace api.Tests.Graph;

public partial class SoftwareDevelopmentCategoriesTests
{
    public class PortfolioCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
    {
        [Fact]
        public async Task PortfolioCategories_Should_Return_Data()
        {
            var executor = await fixture.GetRequestExecutor();

            var result = await executor.ExecuteAsync(
                """
                {
                  portfolioCategories {
                    totalCount
                    nodes {
                      createdAt
                      id
                      internalId
                      title
                      updatedAt
                      version
                      customers {
                        totalCount
                      }
                      portfolioItems {
                        totalCount
                      }
                      technologyCategories {
                        totalCount
                      }
                    }
                  }
                }
                """);

            Func<IQueryResult> fn = result.ExpectQueryResult;
            fn.Should().NotThrow();
            fn().Errors.Should().BeNullOrEmpty();
            result.ToJson().MatchSnapshot();
        }

        [Fact]
        public async Task PortfolioCategoryById_Should_Return_Data()
        {
            var executor = await fixture.GetRequestExecutor();

            var result = await executor.ExecuteAsync(
                """
                {
                  portfolioCategoryById(id: "UG9ydGZvbGlvQ2F0ZWdvcnkKZzM4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5") {
                    createdAt
                    id
                    internalId
                    title
                    updatedAt
                    version
                  }
                }
                """);

            Func<IQueryResult> fn = result.ExpectQueryResult;
            fn.Should().NotThrow();
            fn().Errors.Should().BeNullOrEmpty();
            result.ToJson().MatchSnapshot();
        }
    }
}
