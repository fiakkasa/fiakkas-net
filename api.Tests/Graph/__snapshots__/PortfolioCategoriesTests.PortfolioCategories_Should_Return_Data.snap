{
  "data": {
    "portfolioCategories": {
      "totalCount": 1,
      "items": [
        {
          "createdAt": "2024-01-01T00:00:00.000Z",
          "id": "38e483e4-6961-4b25-88a9-d1d0a5161109",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "customers": {
            "totalCount": 1
          },
          "portfolioItems": {
            "totalCount": 1
          },
          "technologyCategories": {
            "totalCount": 1
          }
        }
      ]
    }
  }
}
